python/ray/dashboard/client/src/pages/dashboard/tune/TuneErrors.tsx

```diff
@@ -0,0 +1,131 @@
+
+const styles = (theme: Theme) =>
+  createStyles({
+    table: {
+      marginTop: theme.spacing(1)
+    },
+    cell: {
+      padding: theme.spacing(1),
+      textAlign: "center",
+      "&:last-child": {
+        paddingRight: theme.spacing(1)
+      }
+    }
+  });
+
+const mapStateToProps = (state: StoreState) => ({
+  tuneInfo: state.dashboard.tuneInfo
+});
+
+const mapDispatchToProps = dashboardActions;
+
+interface State {
+  currentError: string;
+  open: boolean;
+}
+
+class TuneErrors extends React.Component<
+  WithStyles<typeof styles> &
+    ReturnType<typeof mapStateToProps> &
+    typeof mapDispatchToProps,
+  State
+> {
+  state: State = {
+    currentError: "",
+    open: false
+  };
+
+  handleOpen = (key: string) => {
+    this.setState({
+      open: true,
+      currentError: key
+    });
+  };
+
+  handleClose = () => {
+    this.setState({
+      open: false
+    });
+  };
+
+  render() {
+    const { classes, tuneInfo } = this.props;
+    const { currentError, open } = this.state;
+
+    if (tuneInfo === null || Object.keys(tuneInfo["errors"]).length === 0) {
+      return null;
+    }
+
+    return (
+      <div>
+        <Table className={classes.table}>
+          <TableHead>
+            <TableRow>
+              <TableCell className={classes.cell}> Job ID</TableCell>
+              <TableCell className={classes.cell}> Trial ID </TableCell>
+              <TableCell className={classes.cell}> Trial Directory </TableCell>
+              <TableCell className={classes.cell}> Error </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {tuneInfo["errors"] !== null &&
+              Object.keys(tuneInfo["errors"]).map((key, index) => (
+                <TableRow key={index}>
+                  <TableCell className={classes.cell}>
+                    {tuneInfo["errors"][key]["job_id"]}
+                  </TableCell>
+                  <TableCell className={classes.cell}>
+                    {tuneInfo["errors"][key]["trial_id"]}
+                  </TableCell>
+                  <TableCell className={classes.cell}>{key}</TableCell>
+                  <TableCell className={classes.cell}>
+                    <Link
+                      component="button"
+                      variant="body2"
+                      onClick={() => {
+                        this.handleOpen(key);
+                      }}
+                    >
+                      Show Error
+                    </Link>
+                  </TableCell>
+                </TableRow>
+              ))}
+          </TableBody>
+        </Table>
+        <Dialog fullWidth maxWidth="md" open={open} onClose={this.handleClose}>
+          <DialogContent>
+            {open && (
+              <NumberedLines
+                lines={tuneInfo["errors"][currentError]["text"]
+                  .trim()
+                  .split("\n")}
+              />
+            )}
+          </DialogContent>
+        </Dialog>
```

Comment:
Consider using the `DialogWithTitle` component from `common/DialogWithTitle.tsx` for this. See example usage in `pages/dashboard/node-info/NodeInfo.tsx`.

Reply:
Done